pkgs/applications/misc/ebook2cw/default.nix

```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchurl, lame, libvorbis }:
+
+stdenv.mkDerivation rec {
+
+  name = "ebook2cw-${version}";
+  version = "0.8.2";
+
+  src = fetchurl {
+    url = "http://fkurz.net/ham/ebook2cw/${name}.tar.gz";
+    sha256 = "0n2p1i8cnnl2y3cdw8knx1vvs39sid45fr59p6f4vrprmb37l9yq";
+  };
+
+# TODO: Use SVN once fetchsvn stops stripping the trailing slash,
+# otherwise we get a build failure due to a redirect cycle.
+#
+#  src = fetchsvn {
+#    url = "http://svn.fkurz.net/ebook2cw/tags/${name}/";
+#    sha256 = "0n2p1i8cnnl2y3cdw8knx1vvs39sid45fr59p6f4vrprmb37l9yq";
+#  };
+
+  buildInputs = [ lame libvorbis ];
+
+  patchPhase = "patch -p0 < ${./configfile.patch}";
```

Comment:
Append `a/` and `b/` to first two lines of the patch:

```
--- a/ebook2cw.c	2017-11-08 19:52:58.298131348 -0700
+++ b/ebook2cw.c	2017-11-08 19:53:02.588231067 -0700
@@ -136,7 +136,7 @@
 	char isomap[256][4]; 		/* by these strings */
 	char utf8map[256][8];

-	char configfile[1025];
+	char configfile[2048];

 	char id3_author[80],
 		id3_title[80],
```

Then you can just use `patches` list, that would make it easier for other people to apply their own patches:

`patches = [ ./configfile.patch ];`

Reply:
👍